src/pykeen/regularizers.py

```diff
@@ -103,6 +104,41 @@ def forward(self, x: torch.FloatTensor) -> torch.FloatTensor:  # noqa: D102
         return torch.zeros(1, dtype=x.dtype, device=x.device)
 
 
+@functools.lru_cache(maxsize=1)
```

Comment:
what's the purpose of only caching the one most recent?

Reply:
When training, `p` and `d` are fixed. There will only be changes when running HPO, and having multiple trials. In that case, we only need the latest value, since other trials have already been finished.

Reply:
ahh very smart!